### ISSUE TITLE
Bug: [prefer-nullish-coalescing] Empty string check is incorrectly being flagged as an error

### ISSUE THREAD

**Issue by jdufresne** [typescript-eslint/typescript-eslint#10736]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.tsx&code=MYewdgzgLgBBBcMDeMBGIAmBPA-I6ATgJZgDmMAvjALzJqZaIBETlA3AFAA2AprAB404AOnTYYOEWKwxEYAK5cunIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WlqYSNkAC1pkA9gEMkyMswDm6KL2jjokcGAC%2BILUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const s: { body?: string } = { body: "" };
let x = s.body ? s.body : null;
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/prefer-nullish-coalescing": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No error. The string value `s.body` may be an empty string which evaluates to false. By using the `??` operator, the empty string check no longer occurs within the conditional.

### Actual Result

```
Prefer using nullish coalescing operator (`??`) instead of a ternary expression, as it is simpler to read. 2:9 - 2:31

> Fix to nullish coalescing operator (`??`).
```

### Additional Info

Error began to be emitted in 8.22.0.

**Comment by kirkwaiblinger** [comment]
The rule newly reports on ternaries, but using the same conditions as the logical `||` operand, i.e. the following also reports and always has:

[playground](https://typescript-eslint.io/play/#ts=5.7.2&fileType=.tsx&code=MYewdgzgLgBBBcMDeMBGIAmBPA-I6ATgJZgDmMAvjALzJqZaIBETlA3AFAA2AprAB404AOnTYYAHwkwwAVy5dOQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WlqYSNkAC1pkA9gEMkyMswDm6KL2jjokcGAC%2BILUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)
```ts
const s: { body?: string } = { body: "" };
let x = s.body || null;
```

You can control both with [the `ignorePrimitives` option](https://typescript-eslint.io/rules/prefer-nullish-coalescing/#ignoreprimitives):

[Playground](https://typescript-eslint.io/play/#ts=5.7.2&fileType=.tsx&code=MYewdgzgLgBBBcMDeMBGIAmBPA-I6ATgJZgDmMAvjALzJqZaIBETlA3AFAA2AprAB404AOnTYYOEWKwxEYAK5cunXrBm0IohjAA%2BOmAqVsgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WlqYSNkAC1pkA9gEMkyMswDm6MAG1IAd0nQmkADSZw2HJXlNxnAApUAtpXyUAbiiX44iAxAC%2BAXQMeQHoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

### Code

```ts
const s: { body?: string } = { body: "" };
let x = s.body ? s.body : null;
let y = s.body || null;
```

### Eslint config

```json
{
  "rules": {
    "@typescript-eslint/prefer-nullish-coalescing": ["warn", {
      "ignorePrimitives": true
    }]
  }
}
```

### TypeScript config

```json
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `8.22.0` |
| `@typescript-eslint/parser` | `8.22.0` |
| `TypeScript` | `5.7.2` |
| `ESLint` | `9.18.0` |
| `node` | `web` |
